theano/gpuarray/opt.py

```diff
@@ -340,29 +330,28 @@ def apply(self, fgraph):
 
             new_ops = None
             outputs = []
-            ex_opt_time = None
             # Apply the lifter
             for lopt in (self.local_optimizers_all +
                          self.local_optimizers_map.get(type(node.op), []) +
                          self.local_optimizers_map.get(node.op, [])):
-                
-                process_count[lopt] += 1
+
                 if move_to_GPU:
                     t_opt = time.time()
                     try:
-                        new_ops = lopt.transform(
-                            node.op, context_name,
-                            [mapping[i] for i in node.inputs])
+                        new_ops = lopt.transform(node.op, context_name,
+                                                 [mapping[i] for i in node.inputs])
                     except TypeError:
-                        # Updating again because else we'd be counting 
+                        # Updating again because else we'd be counting
                         # time for two except clauses
                         t_opt = time.time()
-                        new_ops = lopt.transform(node.op, context_name, 
-                            [mapping[i] for i in node.inputs],
-                            node.outputs)
+                        new_ops = lopt.transform(node.op, context_name,
+                                                 [mapping[i] for i in node.inputs],
+                                                 node.outputs)
```

Comment:
@abergeron what do you think of passing node.outputs instead of the clients? This allow to get the clients and prevent some optimization from creating the node again to know the output dtype.

If you agree, I would make all optimization to take that extra parameter and remove this try..except.